pkg/contexts/ocm/grammar/grammar_test.go

```diff
@@ -91,6 +91,14 @@ var _ = Describe("ref matching", func() {
 			Expect(VersionRegexp.MatchString("v1-rc.1")).To(BeTrue())
 			Expect(VersionRegexp.MatchString("1.1.1-rc.1")).To(BeTrue())
 		})
+
+		It("matches complex semver", func() {
+			Expect(VersionRegexp.MatchString("0.2.3+2024.T06b")).To(BeTrue())
+		})
+
+		It("matches complex semver with v prefix", func() {
+			Expect(VersionRegexp.MatchString("v0.2.3+2024.T06b")).To(BeTrue())
+		})
```

Comment:
May be we should add tests for all the combinations, also:
- simple Pre
- multi Pre
- simple Build
- multi Build
- simple pre / simple build
- simple pre / multi build
- multi pre / simple build
-  multi pre / multi builds
looks like another table with three possibilities for pre and three for build